core/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -460,4 +460,24 @@ object TreeOps {
   // procedure syntax has decltpe: Some("")
   def isProcedureSyntax(defn: Defn.Def): Boolean =
     defn.decltpe.exists(_.tokens.isEmpty)
+
+  def isCaseClass(tree: Tree): Boolean = {
+    tree.is[Defn.Class] && tree.children.exists(_.is[Mod.Case]) ||
+      tree.parent.exists(_.is[Defn.Class]) && tree.parent.exists(_.children.exists(_.is[Mod.Case]))
+  }
+
+  def isCaseObject(tree: Tree): Boolean = {
+    tree.is[Defn.Object] && tree.children.exists(_.is[Mod.Case]) ||
+      tree.parent.exists(_.is[Defn.Object]) && tree.parent.exists(_.children.exists(_.is[Mod.Case]))
```

Comment:
true, maybe move it to a parameter, since in my limited experience I mostly want to know if a token is part of the definition?

```
  def isCaseClass(tree: Tree, matchOnParent: Boolean = false): Boolean = tree match {
    case defn: Defn.Class => defn.mods.exists(_.is[Mod.Case])
    case _ if matchOnParent => tree.parent.exists(isCaseClass(_))
    case _ => false
  }
```